pkg/apis/openebs.io/v1alpha1/migrationtask.go

```diff
@@ -0,0 +1,124 @@
+/*
+Copyright 2020 The OpenEBS Authors
```

Comment:
Q -> Can client create these crs and start the migration job or It will be created only by the job?

Reply:
Q - what will be the name of the cr for pool migrate and volume migrate?
Do we need to maintain a history if we can execute the job 2 times?

Reply:
1. Either the client will create or if not found when the job starts the job will automatically create one.
2. For the above point to work, the name of the CR should be standardized as `migrate-<type>-<name>`
   examples: 
   `migrate-cstorpool-<spc-name>`
   `migrate-cstorvolume-<pv-name>`

Reply:
Ok.
If job failed for 1st time and client wants to maintain the result and want to start the same job then can we do that?

Reply:
We will clean the previous runs detailed status and keep the latest status only. As the job is idempotent if it failing for some reason that cannot be controlled by the job the status will remain in each run. We also a number of `retries` field which will tell us how many times the job ran.